src/test/java/org/snomed/snowstorm/rest/ConceptControllerTest.java

```diff
@@ -378,4 +379,200 @@ void testSearchAfter() throws ServiceException {
+			CodeSystem codeSystem = new CodeSystem(projectBranch, projectBranch);
+			codeSystemService.createCodeSystem(codeSystem);
+
+			//Add Concepts to branch
+			conceptService.create(
+					new Concept("12345678910")
+							.addDescription(new Description("Computer structure (computer structure)")
+									.setTypeId(Concepts.FSN)
+									.addLanguageRefsetMember(Concepts.US_EN_LANG_REFSET, Concepts.PREFERRED))
+							.addDescription(new Description("Computer structure")
+									.setTypeId(Concepts.SYNONYM)
+									.addLanguageRefsetMember(Concepts.US_EN_LANG_REFSET, Concepts.PREFERRED))
+							.addAxiom(new Relationship(Concepts.ISA, Concepts.SNOMEDCT_ROOT)),
+					projectBranch);
+
+			conceptService.create(
+					new Concept("1234567891011")
+							.addDescription(new Description("Non specific site (computer structure)")
+									.setTypeId(Concepts.FSN)
+									.addLanguageRefsetMember(Concepts.US_EN_LANG_REFSET, Concepts.PREFERRED))
+							.addDescription(new Description("Non specific site")
+									.setTypeId(Concepts.SYNONYM)
+									.addLanguageRefsetMember(Concepts.US_EN_LANG_REFSET, Concepts.PREFERRED))
+							.addAxiom(new Relationship(Concepts.ISA, "12345678910")),
+					projectBranch);
+
+			conceptService.create(
+					new Concept("123456789101112")
+							.addDescription(new Description("Specific site (computer structure)")
+									.setTypeId(Concepts.FSN)
+									.addLanguageRefsetMember(Concepts.US_EN_LANG_REFSET, Concepts.PREFERRED))
+							.addDescription(new Description("Specific site")
+									.setTypeId(Concepts.SYNONYM)
+									.addLanguageRefsetMember(Concepts.US_EN_LANG_REFSET, Concepts.PREFERRED))
+							.addAxiom(new Relationship(Concepts.ISA, "12345678910"), new Relationship("12345", "12345678910"))
+							.addRelationship(new Relationship(Concepts.ISA, "12345678910")),
+					projectBranch);
+
+			//Version project
+			codeSystemService.createVersion(codeSystem, 20200131, "Release 2020-01-31.");
+		};
+		methodTestDataFixture.insert();
+
+		//when
+		ResponseEntity<ConceptHistory> responseEntity = this.restTemplate.exchange(requestUrl, HttpMethod.GET, new HttpEntity<>(null), new ParameterizedTypeReference<ConceptHistory>() {
+		});
+		Map<String, Set<ComponentType>> history = responseEntity.getBody().getHistory();
+		Set<ComponentType> januaryRelease = history.get("20200131");
+
+		//then
+		assertEquals(200, responseEntity.getStatusCodeValue());
+		assertEquals(1, history.size()); //Concept has not changed since first release.
+		assertEquals(4, januaryRelease.size()); //Concept was created with Description, Relationship & Axiom
+		for (ComponentType componentType : januaryRelease) {
+			assertTrue(
+					componentType.equals(ComponentType.Concept) ||
+							componentType.equals(ComponentType.Description) ||
+							componentType.equals(ComponentType.Relationship) ||
+							componentType.equals(ComponentType.Axiom)
+			);
+		}
+	}
+
+	protected interface Procedure {
```

Comment:
I didn't want to add a private method containing all of my "given" logic so I nested it within a "procedure". I think this can help the readability of the test as the "data setup" can be hidden but isn't exposed to other methods.

Reply:
Nice idea and great unit tests 😃